node/node.go

```diff
@@ -2169,40 +2157,32 @@ func (app *App) startSynchronous(ctx context.Context) (err error) {
 	return nil
 }
 
-func (app *App) preserveAfterRecovery(ctx context.Context) {
-	if app.preserve == nil {
+func (app *App) preserveAfterRecovery(ctx context.Context, preserved *checkpoint.PreservedData) {
+	if preserved == nil {
```

Comment:
NIT: since this is already known before the call to this method, maybe we could just do this check before `L#2144`?

Also, this method seem to be able to just error continuously without any consequence - all errors are swallowed and the execution continues. My suggestion/claim is that if an error is swallowed and is recoverable then it is not really an error. So in this case I'm doubting whether these errors should be: 1) logged as `Error`? (maybe just on Debug/Warn) 2) if they are recoverable is it of any use to [display to the end user](https://github.com/spacemeshos/go-spacemesh/blob/develop/CODING.md#users)? 3) logged at all?
This comment is a ditto for all the errors below which are swallowed.

Reply:
>NIT: since this is already known before the call to this method, maybe we could just do this check before L#2144?

:+1: Moved the check there.

Regarding error handling in this function. I'm not 100% sure. I think the author's intent was that this is "best-effort". But that doesn't make much sense because a node that failed to recover (preserve) its own dependencies is unlikely to be able to function properly. So I think it could very well just terminate. @fasmat WDYT?

Reply:
The `preserve` is the data the local node collected it knew about, but wasn't contained in the checkpoint. Failing to restore it won't put the node in an unusable state (the checkpoint has successfully been restored). But data of the node that might be important for it (and wasn't included in the checkpoint) is now gone.

Afaik there is no way to recover from this (unless the node has a backup) but the node should still work normally.